app/controllers/api/v2/table_preferences_controller.rb

```diff
@@ -0,0 +1,55 @@
+module Api
+  module V2
+    class TablePreferencesController < V2::BaseController
+      include Api::Version2
+      include Foreman::Controller::UserAware
+
+      before_action :find_resource, :only => [:destroy, :update, :show]
+
+      api :GET, "/users/:user_id/table_preferences", N_("List of table preferences for a user")
+      def index
+        @table_preferences = @user.table_preferences
+      end
+
+      api :GET, "/users/:user_id/table_preferences/:name", N_("List of table preferences for a table")
+      param :name, String, :required => true
+      def show
+        if @table_preference.blank?
+          @table_preference = TablePreference.new(:user => @user, :name => params[:name])
+        end
+      end
+
+      def_param_group :table_preference do
+        param :table_preferences, Hash, :required => true do
+          param :name, String, :required => true, :desc => N_("name of the table")
+          param :columns, Array, :desc => N_("List of user selected columns")
+        end
+      end
+
+      api :POST, "/users/:user_id/table_preferences/", N_("Creates a table preference for a given table")
+      param_group :table_preference
+      def create
+        @table_preference = @user.table_preferences.build(:name => params[:name], :columns => params[:columns])
+        process_response @table_preference.save
+      end
+
+      api :PUT, "/users/:user_id/table_preferences/:name", N_("Updates a table preference for a given table")
+      param_group :table_preference
+      def update
+        process_response @table_preference.update_attributes(:columns => params[:columns])
+      end
+
+      api :DELETE, "/users/:user_id/table_preferences/:name/", N_("Delete columns for a table")
```

Comment:
Doesn't this delete all the preferences for that table?

Reply:
Nope just the intended preference. I had 2 calls before reset_all and destroy to mitigate this confusion. I was asked to change that and just keep destroy. We likely want to reset things only for a single table/resource name.